### ISSUE TITLE
Check whether `ScaffoldFeatureController` is currently closed

### ISSUE THREAD

**Issue by ghost** [flutter/flutter#39484]
`ScaffoldFeatureController` has the getter `closed`, which returns a `Future<U>` that completes as soon as the feature closes. 

Because of that asynchronous design choice, it's not possible to acknowledge whether the feature tied to the controller is **currently** closed. 

As a practical consequence, the only way to know if a snack-bar is currently opened is by checking whether its duration is greater than the time since it was opened, but this is not a reliable source since the user may have manually dismissed the snack-bar. This problem created a flaw in my tiny library that allows the user to press the back-button twice to close the app: https://github.com/hugocbpassos/double_back_to_close_app/issues/2

I'd suggest you to expose `ScaffoldFeatureController._completer.isCompleted`.

**Comment by Klabauterman** [comment]
You could use a workaround: 

```
class MessageController {
  final ScaffoldFeatureController<SnackBar, SnackBarClosedReason> _controller;
  bool closed = false;

  MessageController(this._controller) {
    init();
  }

  init() async {
    await _controller.closed;
    closed = true;
  }

  dismiss() {
    if (closed) return;
    _controller.close();
  }
}
```

**Comment by TechAurelian2** [comment]
I also agree it would be very useful to know directly if a snack-bar is currently opened.

What I'm using as a workaround:

```dart
class _InspectionScreenState extends State<InspectionScreen> {
  /// The controller for the snack bar that displays the inspection screen tip.
  ScaffoldFeatureController<SnackBar, SnackBarClosedReason>? _tipController;
  
  @override
  void dispose() {
    // Remove the tip snack bar if it is still visible
    if (_tipController != null) _tipController!.close();

    super.dispose();
  }
  
  ...
  
  _tipController = ScaffoldMessenger.of(context).showSnackBar(snackBar);
  _tipController!.closed.then((_) => _tipController = null);
```

**Comment by RC0D3** [comment]
Worst design ever.

I created simple ToDo App, and when my user clear 3 or more tasks, I got bug in snackbars, never delete.

How it occours:
My user drag an task, and close, then call showSnackBar.


Then if I try to removeCurrentSnackBar, this will throw animation flick, new animation for every delete, UI change height for every delete, so bad for UX.

Then I use hideCurrentSnackBar, for 2 snack works fine, but some way, if 3 or more, this lost some snack from Queue, and never disappear!

If I try clearSnackBars same UX problem.


I hate how this snackBar work, no managment, it only throw away.
